### PR TITLE
Chore: Fix release script

### PR DESCRIPTION
**Pull request by antross**
## Pull request checklist

Make sure you:

- [x] Signed the Contributor License Agreement (after creating PR)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [ ] Added/Updated related documentation.
- [ ] Added/Updated related tests.

### CODE REVIEW

**Comment by antross** [comment]
@flynnolivia you should be able to test the basics of this locally using `npm run release -- --dry-run`

File: release/lib/git-helpers.ts [hunk 1]
```diff
@@ -89,21 +95,19 @@ const getResponseForCommitInfoRequest = async (commitSHA: string) => {
...
-            debug(`Reusing cached Author ${commitInfo.author?.login}`);
+            debug(`Reusing cached Author ${commitInfo.author.login}`);
 
             return cachedAuthor;
         }
 
-        debug(`Requesting info for login ${commitInfo.author?.login}`);
+        debug(`Requesting info for login ${commitInfo.author.login}`);
 
         const responseForUserInfoRequestPromise = getResponseForUserInfoRequest(commitInfo);
```
Comment (antross) [inline release/lib/git-helpers.ts:108]: Note: This call specifically didn't expect author to be `null`, we just hadn't hit it in a previous release. The GitHub libraries were updated to allow `null` ~8 months ago. I updated other cases flagged by TypeScript then but missed this one. Fixed by making the `null` handling an early return instead.

File: scripts/utils/exec.js [hunk 2]
```diff
@@ -21,22 +21,27 @@ const exec = (cmd, options = {}) => {
...
         command.on('exit', (code) => {
-            if (code !== 0) {
-                return reject(new Error('NoExitCodeZero'));
+            if (code) {
+                reject(new Error(`Exit Code: ${code}\n${stdout}`));
+            } else if (!command.stdout) {
+                resolve({ stdout: '' });
             }
-
-            return resolve({ stdout: stdout.trimEnd() });
```
Comment (antross) [inline scripts/utils/exec.js:37]: Per https://nodejs.org/api/child_process.html#event-exit:
> When the `'exit'` event is triggered, child process stdio streams might still be open.

Hence when `command.stdout` exists we now wait for the stream to close before resolving.

**Comment by molant** [comment]
@antross there used to be a problem with the release script and certain node versions, right? Did it ever got fixed?

**Comment by antross** [comment]
@molant yes there was issue with it hanging using the standard command prompt on Windows as I recall. I've switched to running it via WSL and haven't checked that flow in awhile, though.

**Review by flynnolivia — APPROVED** [review]

**Pull request merged**